src/mappings/bondingManager.ts

```diff
@@ -259,9 +315,12 @@ export function rebond(event: Rebond): void {
   delegator.lastClaimRound = round.id;
   delegator.bondedAmount = convertToDecimal(delegatorData.value0);
   delegator.fees = convertToDecimal(delegatorData.value1);
-  delegator.unbonded = delegator.unbonded.minus(
-    convertToDecimal(event.params.amount)
-  );
+
+  if (delegator.unbonded.gt(ZERO_BD)) {
```

Comment:
I think a problem with this check is that it will pass in the scenario where the `unbonded > 0` *and* the delegator is the receiver for a previous `TransferBond` event which would result in subtracting the `TransferBond` amount from `unbonded`. This could still result in a negative value for `unbonded` or `unbonded` being incorrectly decreased - IIUC `unbonded` should only be decreased if the delegator rebonds an unbonding lock that is previously created since the creation of the unbonding lock would've increased `unbonded`.

Could we check for the existence of a previous `Unbond` event for the delegator and the unbonding lock ID in order to determine if we should subtract from `unbonded` i.e. looking up if there is a `Unbond` event with the delegator's address and the unbonding lock ID?

Reply:
Oh yea good spot. So this would occur if a delegator bonds and unbonds a partial amount on L2 before migrating their stake, is that right?

Scenario:
1. Orchestrator bonds 10 LPT on L2
2. Orchestrator unbonds 2 LPT on L2 (`unbonded` now equals 2)
3. Orchestrator migrates 20 LPT on from L1 to L2
4. The rebond event is triggered subtracting 20 from 2 which results in a negative `unbonded` value

> Could we check for the existence of a previous Unbond event for the delegator and the unbonding lock ID in order to determine if we should subtract from unbonded i.e. looking up if there is a Unbond event with the delegator's address and the unbonding lock ID?

We check for the unbonding lock emitted in the rebond event associated with the delegator [here](https://github.com/livepeer/subgraph/blob/1defad4a0393f5071be14c67aa6e7f5b525fe4c6/src/mappings/bondingManager.ts#L336). If we move the subtraction inside this conditional, would that solve the problem?

Reply:
> So this would occur if a delegator bonds and unbonds a partial amount on L2 before migrating their stake, is that right?

Yep. More generally, this would happen if the delegator calls `transferBond()` after unbonding a partial amount on L2 (stake migration is a case where `transferBond()` is called).

> Scenario:

Yep this scenario checks out.

> If we move the subtraction inside this conditional, would that solve the problem?

This wouldn't work because the unbonding lock in that link would exist in the scenario you presented as well because it would be created by the `transferBond` handler. 

Another option that I think could work is adding a `sender` field to the `UnbondingLock` entity which is set to the delegator's address in the `unbond` handler and set to `oldDelegator` in the `transferBond` handler. Then, in `rebond` you could check if the `sender` field for the lock is equal to the delegator's address - if it is then we know that this is an unbonding lock the delegator created by calling `unbond()` and if it is not then we know that this is an unbonding lock created by `transferBond()`. In the latter case, we wouldn't need to subtract from `unbonded`.

Reply:
> Another option that I think could work is adding a sender field to the UnbondingLock entity which is set to the delegator's address in the unbond handler and set to oldDelegator in the transferBond handler. Then, in rebond you could check if the sender field for the lock is equal to the delegator's address - if it is then we know that this is an unbonding lock the delegator created by calling unbond() and if it is not then we know that this is an unbonding lock created by transferBond(). In the latter case, we wouldn't need to subtract from unbonded.

Makes sense, thanks. Just pushed up the above fix.